crates/stackable-webhook/src/webhooks/mutating_webhook.rs

```diff
@@ -0,0 +1,217 @@
+///
+/// ```
+/// use std::sync::Arc;
+///
+/// use k8s_openapi::api::admissionregistration::v1::MutatingWebhookConfiguration;
+/// use k8s_openapi::api::apps::v1::StatefulSet;
+///
+/// use stackable_operator::builder::meta::ObjectMetaBuilder;
+/// use stackable_operator::kube::Client;
+/// use stackable_operator::kube::core::admission::{AdmissionRequest, AdmissionResponse};
+/// use stackable_operator::kvp::Label;
+/// use stackable_webhook::WebhookServer;
+/// use stackable_webhook::webhooks::MutatingWebhook;
+///
+/// # async fn docs() {
+/// // The Kubernetes client
+/// let client = Client::try_default().await.unwrap();
+/// // The context of the controller, e.g. contains a Kubernetes client
+/// let ctx = Arc::new(());
+/// // Read in from user input, e.g. CLI arguments
+/// let disable_restarter_mutating_webhook = false;
+///
+/// let mutating_webhook = Box::new(MutatingWebhook::new(
+///     get_mutating_webhook_configuration(),
+///     my_handler,
+///     ctx,
+///     disable_restarter_mutating_webhook,
+///     client,
+///     "my-field-manager".to_owned(),
+/// ));
+///
+/// let webhook_options = todo!();
+/// let webhook_server = WebhookServer::new(webhook_options, vec![mutating_webhook]).await.unwrap();
+/// webhook_server.run().await.unwrap();
+/// # }
+///
+/// fn get_mutating_webhook_configuration() -> MutatingWebhookConfiguration {
+///     let webhook_name = "pod-labeler.stackable.tech";
+///
+///     MutatingWebhookConfiguration {
+///         webhooks: Some(vec![k8s_openapi::api::admissionregistration::v1::MutatingWebhook {
+///             // This is checked by the stackable_webhook code
+///             admission_review_versions: vec!["v1".to_owned()],
+///             ..Default::default()
+///         }]),
+///         ..Default::default()
+///     }
+/// }
+///
+/// // Basic no-op implementation
+/// pub async fn my_handler(
+///     ctx: Arc<()>,
+///     request: AdmissionRequest<StatefulSet>,
+/// ) -> AdmissionResponse {
+///     AdmissionResponse::from(&request)
+/// }
+/// ```
+pub struct MutatingWebhook<H, S, R> {
+    mutating_webhook_configuration: MutatingWebhookConfiguration,
+    handler: H,
+    handler_state: Arc<S>,
+    _resource: PhantomData<R>,
+
+    disable_mutating_webhook_configuration_maintenance: bool,
+    client: Client,
+
+    /// The field manager used when maintaining the MutatingWebhookConfigurations.
+    field_manager: String,
+}
+
+impl<H, S, R> MutatingWebhook<H, S, R> {
+    /// All webhooks need to set the admissionReviewVersions to `["v1"]`, as this mutating webhook
+    /// only supports that version! A failure to do so will result in a panic.
+    ///
+    /// Your [`MutatingWebhookConfiguration`] can contain 0..n webhooks, but it is recommended to
+    /// only have a single entry in there, as the clientConfig of all entries will be set to the
+    /// same service, port and HTTP path.
+    pub fn new(
+        mutating_webhook_configuration: MutatingWebhookConfiguration,
```

Comment:
question: Should we maybe construct this on our own (opinionated) way? From the looks of it, it seems fairly painful to use...